lib/src/layout/scaffold.dart

```diff
@@ -1,118 +1,331 @@
+                      decoration: BoxDecoration(
+                        color: backgroundColor,
+                      ).copyWith(
+                        color: widget.titleBar?.decoration?.color,
+                        image: widget.titleBar?.decoration?.image,
+                        border: widget.titleBar?.decoration?.border ??
+                            Border(bottom: BorderSide(color: dividerColor)),
+                        borderRadius: widget.titleBar?.decoration?.borderRadius,
+                        boxShadow: widget.titleBar?.decoration?.boxShadow,
+                        gradient: widget.titleBar?.decoration?.gradient,
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+
+            // Sidebar resizer
+            if ((widget.sidebar?.isResizable ?? false) && canShowSidebar)
+              Positioned(
+                left: _sidebarWidth - 4,
+                width: 7,
+                height: height,
+                child: GestureDetector(
+                  behavior: HitTestBehavior.opaque,
+                  onHorizontalDragUpdate: (details) {
+                    setState(() {
+                      _sidebarWidth = math.max(
+                        widget.sidebar!.minWidth,
+                        math.min(
+                          math.min(widget.sidebar!.maxWidth!, width),
+                          _sidebarWidth + details.delta.dx,
+                        ),
+                      );
+                    });
+                  },
+                  child: MouseRegion(
+                    cursor: SystemMouseCursors.resizeColumn,
+                    child: Align(
+                      alignment: Alignment.center,
+                      child: VerticalDivider(
+                        thickness: 1,
+                        width: 1,
+                        color: dividerColor,
+                      ),
                     ),
-                    view2: body,
-                    viewMode: SplitViewMode.Horizontal,
-                  );
-                } else {
-                  return body;
-                }
-              },
-            )
-          : body,
+                  ),
+                ),
+              ),
+          ],
+        );
+
+        return ValueListenableBuilder<Map<Key, double>>(
+          child: layout,
+          valueListenable: _valueNotifier,
+          builder: (_, panes, child) {
+            double sum = panes.values.fold(0.0, (prev, curr) => prev + curr);
+            double _remainingWidth = width - (visibleSidebarWidth + sum);
+
+            return ScaffoldScope(
+              child: child!,
+              scaffoldState: this,
+              constraints: constraints,
+              valueNotifier: _valueNotifier,
+              contentAreaWidth: math.max(_minContentAreaWidth, _remainingWidth),
+            );
+          },
+        );
+      },
     );
   }
 }
+
+class ScaffoldScope extends InheritedWidget {
```

Comment:
This class needs documentation. The class members and functions need docs as well.

Reply:
This can be done later